services/device-registry-mongodb/src/main/java/org/eclipse/hono/deviceregistry/mongodb/model/BaseDto.java

```diff
@@ -14,23 +14,25 @@
 
 import java.time.Instant;
 import java.util.Objects;
+import java.util.Optional;
 
 import org.eclipse.hono.annotation.HonoTimestamp;
 import org.eclipse.hono.util.RegistryManagementConstants;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 /**
- * TODO.
+ * The base class for implementing a DTO (Data Transfer Object) to store data in mongodb.
  */
 @JsonIgnoreProperties(ignoreUnknown = true)
 @JsonInclude(JsonInclude.Include.NON_NULL)
 public abstract class BaseDto {
 
     @JsonProperty(value = RegistryManagementConstants.FIELD_VERSION, required = true)
-    protected String version;
+    protected Long versionNo;
```

Comment:
is it helpful to use a Long here instead of a String, i.e. does the Mongo DB driver update this field automatically or do you need to set this value manually whenever you save a modified object instance?

Reply:
Since it is a sequence number, I thought its safe and better to use the `Long` instead of storing as a `String`. Mongo DB driver update this field automatically based on the value store in the database. But the version number is being incremented manually before the database update.

Reply:
I agree and I wonder why we have a *getVersion* method that returns a String. Do we need such a method?
Does the `null` value have any special semantics for this property?

Reply:
In many device registration operations, `Future<OperationResult>` is being returned. When an operation is successful `OperationResult.ok(...)` is being used for that and it expects the version to be String. 

```
OperationResult<T> ok(final int status, final T payload,
            final Optional<CacheDirective> directive, final Optional<String> resourceVersion)
```
For that purpose, `getVersion()` is needed and it returns a `String`.

Reply:
> Does the `null` value have any special semantics for this property?

Actually not, since we always store the version eventhough the incoming requests do not contain any version.